request.go

```diff
@@ -116,6 +132,10 @@ func NewRequest(opts RequestOptions) (*Request, error) {
 	if err != nil {
 		return nil, &MalformedRequestError{Msg: err.Error()}
 	}
+	hashAlgParameter := opts.HashAlgorithmParameters
+	if reflect.DeepEqual(hashAlgParameter, asn1.RawValue{}) || reflect.DeepEqual(hashAlgParameter, asn1.NullRawValue) {
```

Comment:
What's the difference between above and the code below?
```go
if hashAlgParameter == asn1.RawValue{} {
```

Reply:
`asn1.RawValue{}`, `asn1.NullRawValue` and 
`asn1.RawValue{
     Tag: 5,
     FullBytes: []bytes{5, 0}  // tag and length
}`
are three different structs, where only the 3rd one is a valid asn1 object.